### PR TITLE
Add representers

### PR DESCRIPTION
**Pull request by iHiD**
Added some files/copy about representers. 

@SleeplessByte You may need to update links on the maintainers dashboard for this?

### CODE REVIEW

File: docs/representers/interface.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# The Interface
+
+All interactions with the Exercism website are handled automatically. Representers have the single responsibility of taking a solution and returning a representation of it.
```
Comment (tehsphinx) [inline docs/representers/interface.md:3]: Maybe explain a bit more what a representation is:

```suggestion
All interactions with the Exercism website are handled automatically. Representers have the single responsibility of taking a solution and returning a representation of it.

A _representation_ is an extraction of the solution to its essence without names, comments, spacing, etc. but still uniquely identifying the approach taken. Two different ways of solving the exercise must not have the same representation.
```

**Review by tehsphinx — APPROVED** [review]

File: docs/representers/interface.md (hunk 1 above)
Comment (ccare) [reply docs/representers/interface.md:3]: I think what defines 'essence' here is key and probably flexes a little language by language. I'm still puzzled whether two solutions in, say, Ruby should have the same representation regardless of names.

In the following three code snippets, I would imagine that the first two could safely have the same representation. But with the third, I would imagine a mentor would reject the solution based on the variable name being camelCase and therefore not idiomatic. If so, the same feedback wouldn't apply, and the solutions shoudn't share a single representation.

From my understanding a representation may uniquely identify an approach, but a single approach may have many acceptable representations? 

```ruby
def foo
  resp = "x"
  puts resp
end
```

```ruby
def foo
  # comment here
  my_var = "x"
  puts my_var
end
```

```ruby
def foo
  _myVar = "x"
  puts _myVar
end
```

Comment (tehsphinx) [reply docs/representers/interface.md:3]: I don't think we can account for naming conventions or comments (e.g. in Go) in the representers. If we'd do that we risk rendering them ineffective. They would be too tightly coupled to the specific implementations.

Checking proper naming of variables is something a static analyzer is good at. I think we need to find a way to do both and combine them to get the best of both.

Comment (iHiD) [reply docs/representers/interface.md:3]: Naming conventions can/will be checked by analyzers after we've looked for a representation.

So we do:
- Get representation
- Lookup match in db (get status/comments)
- Give status/comments to representer, which does final check (linting/naming, etc).

That last stage will be optional for tracks to implement.

**Pull request merged**